lib/sycamore/sycamore/tests/integration/connectors/opensearch/test_opensearch_read.py

```diff
@@ -153,51 +210,33 @@ def doc_reconstructor(index_name: str, doc_id: str) -> Document:
-            os_client.indices.refresh(TestOpenSearchRead.INDEX)
+        os_client.indices.refresh(TestOpenSearchRead.INDEX)
 
-        retrieved_docs = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS, index_name=TestOpenSearchRead.INDEX
-        )
-        target_doc_id = original_docs[-1].doc_id if original_docs[-1].doc_id else ""
-        query = {"query": {"term": {"_id": target_doc_id}}}
-        query_docs = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS, index_name=TestOpenSearchRead.INDEX, query=query
-        )
+        expected_count = len(original_docs)
+        actual_count = get_doc_count(os_client, TestOpenSearchRead.INDEX)
+        # refresh should have made all ingested docs immediately available for search
+        assert actual_count == expected_count, f"Expected {expected_count} documents, found {actual_count}"
 
-        retrieved_docs_reconstructed = context.read.opensearch(
-            os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS,
-            index_name=TestOpenSearchRead.INDEX,
-            reconstruct_document=True,
-            doc_reconstructor=doc_reconstructor,
+        retrieved_docs_reconstructed = (
+            context.read.opensearch(
+                os_client_args=TestOpenSearchRead.OS_CLIENT_ARGS,
+                index_name=TestOpenSearchRead.INDEX,
+                reconstruct_document=True,
+                doc_reconstructor=OpenSearchDocumentReconstructor(TestOpenSearchRead.INDEX, doc_reconstructor),
+            )
+            .filter(lambda doc: doc.doc_id == "1")
+            .take_all()
         )
-        original_materialized = sorted(original_docs, key=lambda d: d.doc_id)
-
-        # hack to allow opensearch time to index the data. Without this it's possible we try to query the index before
-        # all the records are available
-        time.sleep(1)
-        retrieved_materialized = sorted(retrieved_docs.take_all(), key=lambda d: d.doc_id)
-        query_materialized = query_docs.take_all()
-        retrieved_materialized_reconstructed = sorted(retrieved_docs_reconstructed.take_all(), key=lambda d: d.doc_id)
 
-        # with OpenSearch(**TestOpenSearchRead.OS_CLIENT_ARGS) as os_client:
-        #    os_client.indices.delete(TestOpenSearchRead.INDEX)
-        assert len(query_materialized) == 1  # exactly one doc should be returned
-        compare_connector_docs(original_materialized, retrieved_materialized)
-
-        assert len(retrieved_materialized_reconstructed) == 1
-        doc = retrieved_materialized_reconstructed[0]
-        assert len(doc.elements) == len(retrieved_materialized) - 1  # drop the document parent record
-
-        for i in range(len(doc.elements) - 1):
-            assert doc.elements[i].element_index < doc.elements[i + 1].element_index
+        assert len(retrieved_docs_reconstructed) == 1
+        assert retrieved_docs_reconstructed[0].data["hidden"] == hidden
```

Comment:
Does `assert retrieved_docs_reconstructed == original_docs` not work?
That's the condition that we want to be true.

Reply:
Yes, I will add this check (`original_docs` is after `.explode()`, but you meant `docs`).